theano/compile/tests/test_function_module.py

```diff
@@ -239,6 +239,41 @@ def test_copy(self):
         f(1, 2)  # put them out of sync
         self.assertFalse(f(1, 2) == g(1, 2))  # they should not be equal anymore.
 
+    def test_copy_share_memory(self):
```

Comment:
Well, I just add two SharedVariable into the test case and give one of them an update. I can separate the test if you want, but that will produce repeated codes.